SCSH_3C_iOS/SCSH_3C_iOS/Modules/Explore/View/DistributorCollectionViewCell.swift

```diff
@@ -0,0 +1,265 @@
+            exploreButton.bottomAnchor.constraint(equalTo: bottomContainerView.bottomAnchor, constant: -10),
+            exploreButton.heightAnchor.constraint(equalTo: likeButton.heightAnchor),
+            
+            likeButton.heightAnchor.constraint(equalToConstant: 30),
+            likeButton.widthAnchor.constraint(equalTo: likeButton.heightAnchor, multiplier: 2),
+            likeButton.trailingAnchor.constraint(equalTo: nameLabel.trailingAnchor),
+            likeButton.bottomAnchor.constraint(equalTo: exploreButton.bottomAnchor),
+        ])
+    }
+    
+    private func setupAction() {
+        likeButton.addTarget(self, action: #selector(likeButtonDidTap), for: .touchUpInside)
+        exploreButton.addTarget(self, action: #selector(exploreButtonDidTap), for: .touchUpInside)
+        
+        mainProductImageView.addGestureRecognizer(mainProductTapGestureRecognizer)
+        subProductImageView_1.addGestureRecognizer(subProduct1TapGestureRecognizer)
+        subProductImageView_2.addGestureRecognizer(subProduct2tapGestureRecognizer)
+        
+        mainProductTapGestureRecognizer.addTarget(self, action: #selector(mainProductImageDidTap))
+        subProduct1TapGestureRecognizer.addTarget(self, action: #selector(subProduct1ImageDidTap))
+        subProduct2tapGestureRecognizer.addTarget(self, action: #selector(subProduct2ImageDidTap))
+    }
+    
+    @objc
+    private func likeButtonDidTap(_ sender: UIButton) {
+        sender.isSelected.toggle()
+        
+        sender.tintColor = sender.isSelected ? .red : .lightGray
+        
+        delegate?.distributorLikeButtonDidTap(distributor: distributor)
+    }
+    
+    @objc
+    private func exploreButtonDidTap(_ sender: UIButton) {
+        sender.isSelected.toggle()
+        
+        delegate?.distributorExploreButtonDidTap(distributor: distributor)
+    }
+    
+    @objc
+    private func mainProductImageDidTap() {
+        guard let productId = distributor.products.first?.id else { return }
+        
+        delegate?.distributorMainProductGestureDidTap(productId: productId)
+    }
+
+    @objc
+    private func subProduct1ImageDidTap() {
+        guard distributor.products.count > 1 else { return }
+
+        delegate?.distributorSubProduct2GestureDidTap(productId: distributor.products[1].id)
+    }
+
+    @objc
+    private func subProduct2ImageDidTap() {
+        guard let productId = distributor.products.last?.id else { return }
+        
+        delegate?.distributorSubProduct2GestureDidTap(productId: productId)
+    }
+    
+    func setup(distributor: Distributor) {
+        self.distributor = distributor
+        
+        mainProductImageView.kf.setImage(with: self.distributor.products.first?.image, placeholder: UIImage(systemName: "photo.fill"))
+        subProductImageView_1.kf.setImage(with: self.distributor.products[1].image, placeholder: UIImage(systemName: "photo.fill"))
```

Comment:
What if the size of `self.distributor.products` is less than 2?

Reply:
I forgot it !! index out of exception... I'll fix it,

Reply:
https://github.com/No-57/SCSH_3C_iOS/pull/42/commits/659bb20c6b0fc1666c688207754a029fdc3021ae